models/classes/runner/RunnerParamParserTrait.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA ;
+ */
+
+namespace oat\taoQtiTest\models\runner;
+
+use oat\taoQtiTest\models\runner\map\QtiRunnerMap;
+use taoQtiTest_helpers_TestRunnerUtils as TestRunnerUtils;
+
+trait RunnerParamParserTrait
+{
+    /**
+     * @var QtiRunnerServiceContext
+     */
+    protected $serviceContext;
+
+    /**
+     * Check if the parameter associated to $name exists
+     *
+     * @param $name
+     * @return mixed
+     */
+    abstract public function hasRequestParameter($name);
```

Comment:
Is it the right place for that? I mean isn't it better to implement an interface instead of using a trait to set abstracts?